rllib/agents/trainer.py

```diff
@@ -363,21 +360,30 @@ def setup(self, config: PartialTrainerConfigDict):
         # parallel to training.
         # TODO: Deprecate `_init()` and remove this try/except block.
         except NotImplementedError:
-            # Only if user did not override `_init()`:
+            _init = False
+
+        # Only if user did not override `_init()`:
+        if not _init:
             # - Create rollout workers here automatically.
             # - Run the execution plan to create the local iterator to `next()`
             #   in each training iteration.
             # This matches the behavior of using `build_trainer()`, which
             # has been deprecated.
-            self.workers = WorkerSet(
-                env_creator=self.env_creator,
-                validate_env=self.validate_env,
-                policy_class=self.get_default_policy_class(self.config),
-                trainer_config=self.config,
-                num_workers=self.config["num_workers"],
-                local_worker=True,
-                logdir=self.logdir,
-            )
+            try:
+                self.workers = WorkerSet(
+                    env_creator=self.env_creator,
+                    validate_env=self.validate_env,
+                    policy_class=self.get_default_policy_class(self.config),
+                    trainer_config=self.config,
+                    num_workers=self.config["num_workers"],
+                    local_worker=True,
+                    logdir=self.logdir,
+                )
+            except RayActorError as e:
+                if e.actor_init_failed:
+                    raise e.args[0].args[2]
```

Comment:
any chance you can help comment a bit what we get from args[0].args[2]?
I can't believe that is a public API from RayActorError ... :)

Reply:
Ah, haha, yeah, great point, will fix.

Reply:
done

Reply:
done